tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -422,6 +424,14 @@ private int WriteTags(ref byte[] bytes, int offset, in SpanModel model, ITagProc
                 }
             }
 
+            // add _dd.base_service tag to spans where the service name has been overrideen
+            if (!serviceNameEqualsDefault && !string.IsNullOrEmpty(model.TraceChunk.DefaultServiceName))
+            {
+                count++;
+                offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _baseServiceNameBytes);
+                offset += MessagePackBinary.WriteString(ref bytes, offset, model.TraceChunk.DefaultServiceName);
```

Comment:
We should try to cache the raw MessagePack bytes for `model.TraceChunk.DefaultServiceName` since they will almost always be the same for the lifetime of the process. See what we do with `versionRawBytes` above:
```csharp
var versionRawBytes = MessagePackStringCache.GetVersionBytes(model.TraceChunk.ServiceVersion);
```

Reply:
This comment is still valid. Note we don't want to cache the bytes _forever_ like `_baseServiceNameBytes`, since it technically can change via config, but it is very rare. `MessagePackStringCache` checks if the string has changed since the last time it was used.

Reply:
> Note we don't want to cache the bytes forever like _baseServiceNameBytes, since it technically can change via config

If it changes via config, don't we reinitialize, and so would have a new `SpanMessagePackFormatter`? 🤔 If so, then we _could_ cache it "forever" I think (per instance, as we do with everything currently)

Reply:
Thank you both, totally missed this comment!

I've added it to be cached.